File: java/JavaFileutil/src/JavaFileutil/App/CommandInterpreter.java
package JavaFileutil.App;

import java.util.Arrays;
import java.util.Scanner;
import java.util.ArrayList;
import java.util.List;
import JavaFileutil.JavaCommandEngine.CommandFactory;

public class CommandInterpreter
{
    private void runEventLoop()
    {
        while (true)
        {
            System.out.print("===> ");
            Scanner scanner = new Scanner(System.in);
            String choice = scanner.nextLine();
            List<String> arguments = new ArrayList<>(Arrays.asList(choice.split("\\s+")));
            arguments.remove("");
            new CommandFactory(arguments).Create().Execute();
        }
    }

    public CommandInterpreter()
    {
    }

    public void run()
    {
        System.out.println("Welcome to the file utility command interpreter!");
        System.out.println("Author: Oleksii <DarkSpectre> Kshenskyi.");
        new CommandFactory(new ArrayList<>(Arrays.asList("usage"))).Create().Execute();
        System.out.println("Type 'usage <command>' to learn about the <command> you're interested in.");
        runEventLoop();
    }
}

File: python/App/CommandInterpreter.py
from PythonCommandEngine.CommandFactory import *


class CommandInterpreter(object):
    @classmethod
    def run_event_loop(cls):
        while True:
            choice = input("===> ")
            arguments = list(filter(None, choice.split(" ")))
            CommandFactory(arguments).create()(arguments).execute()

    @classmethod
    def run(cls):
        print("Welcome to the file utility command interpreter!")
        print("Author: Oleksii <DarkSpectre> Kshenskyi.")
        CommandFactory(["usage"]).create()(["usage"]).execute()
        print("Type 'usage <command>' to learn about the <command> you're interested in.")
        cls.run_event_loop()
